Fix remove unnecessary overview headings (628) (#632)
astronomer/docs: astro/secrets-backend.md

## Secrets Backend Tool Integration Benefits

Integrating a secrets backend tool with Astro allows you to:

- Store Airflow variables and connections in a centralized location alongside secrets from other tools and systems used by your organization, including Kubernetes secrets, SSL certificates, and more.
- Comply with internal security postures and policies that protect your organization.
- Recover in the case of an incident.
- Automatically pull Airflow variables and connections that are already stored in your secrets backend when you create a new Deployment instead of having to set them manually in the Airflow UI.

> Astro supports integration with a variety of secrets backend tools including::

Reviewer: ```suggestion
Astro integrates with the following secret backend tools:
```

"including" implies that there are more, but this is actually the full list

- Hashicorp Vault
- AWS Systems Manager Parameter Store
- AWS Secrets Manager
- Google Cloud Secret Manager
- Azure Key Vault

> Secrets backend integrations are applied individually to each Astro Deployment.

Reviewer: ```suggestion
Secrets backend integrations are configured individually with each Astro Deployment.
```
